tidy3d/components/geometry/base.py

```diff
@@ -158,10 +174,32 @@ def inside_meshgrid(
         return is_inside
 
     @abstractmethod
+    def intersections_tilted_plane(
+        self, normal: Coordinate, origin: Coordinate, to_2D: MatrixReal4x4
+    ) -> List[Shapely]:
+        """Return a list of shapely geometries at the plane specified by normal and origin.
+
+        Parameters
+        ----------
+        normal : Coordinate
+            Vector defining the normal direction to the plane.
+        origin : Coordinate
+            Vector defining the plane origin.
+        to_2D : MatrixReal4x4
```

Comment:
quick question: would it make sense to leave this optional? for example, if not specified, don't perform any transformation?

Reply:
We have to, somehow, map the 3D coordinates to 2D. We could leave an identity there as default, which means using (x,y) from the output directly.

Reply:
ok I see now. let's leave it as is I think.